Observer/Checkout/ValidateCustomer.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Paytrail\PaymentService\Observer\Checkout;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Validation\ValidationException;
+use Paytrail\PaymentService\Model\Validation\PreventAdminActions;
+
+class ValidateCustomer implements ObserverInterface
+{
+    /**
+     * @var PreventAdminActions
+     */
+    private PreventAdminActions $preventAdminActions;
+
+    /**
+     * @param PreventAdminActions $preventAdminActions
+     */
+    public function __construct(
+        PreventAdminActions $preventAdminActions
+    ) {
+        $this->preventAdminActions = $preventAdminActions;
+    }
+
+    public function execute(Observer $observer)
+    {
+        if ($this->preventAdminActions->isAdminAsCustomer()) {
```

Comment:
This prevents the entire checkout from loading, while very effective it also might be a bit too harsh from a vendor module. Could you instead modify the checkout config provider at \Paytrail\PaymentService\Model\ConfigProvider::getConfig and prevent it from supplying the payment method configuration to checkout page if admin user is detected. 

Instead provide the same error message as below.

So admin user would see the address form and shippping methods. But when they go to payment page, they see no payment methods and only see an error message "Admin user is not authorized for this operation"

Reply:
Couldn't use getConfig(), because it runs before landing on checkout shipping page.
I wrote a before plugin for savePaymentInformation() which displays exception when admin as customer want to 'place order'. So admin basically can do things on checkout pages, but can't place an order with any of chosen payment method.